test/integration/change_edition_type_test.rb

```diff
@@ -1,261 +1,93 @@
-
-    click_on "Create as new Answer edition"
-
-    assert page.has_content?(transaction.title)
-    assert page.has_content?("New edition created")
-    assert_field_contains(transaction.more_information, "Body")
-    assert_field_contains(transaction.alternate_methods, "Body")
-  end
-
-  test "should be able to convert a SimpleSmartAnswerEdition into an AnswerEdition" do
-    simple_smart_answer = FactoryGirl.create(:simple_smart_answer_edition, state: 'published')
-    visit_edition simple_smart_answer
-
-    within "div.tabbable" do
-      click_on "Admin"
-    end
-
-    assert page.has_button?("Create as new Answer edition")
-
-    click_on "Create as new Answer edition"
-
-    assert_text(simple_smart_answer.title)
-    assert_text("New edition created")
-  end
-
-# tests for changing Answer, Guide, Programme into a Transaction
+  edition_types = [:transaction_edition, :place_edition, :programme_edition, :simple_smart_answer_edition, :business_support_edition, :answer_edition, :guide_edition, :licence_edition, :video_edition, :completed_transaction_edition, :help_page_edition, :campaign_edition]
 
-  test "should be able to convert an AnswerEdition into a TransactionEdition" do
-    answer = FactoryGirl.create(:answer_edition, state: 'published')
-    visit_edition answer
+  sample_parts = Set.new([{"title" => "PART !", "body" => "This is some edition version text.", "slug" => "part-one"}, {"title" => "PART !!", "body" => "This is some more edition version text.", "slug" =>  "part-two"}])
 
-    within "div.tabbable" do
-      click_on "Admin"
-    end
+  conversions = edition_types.permutation(2).reject { |pair| pair[0] == pair[1] }
 
-    assert page.has_button?("Create as new Transaction edition")
+  conversions.each do |pair|
 
-    click_on "Create as new Transaction edition"
+    from, to = pair
+    should "be able to convert #{from} into #{to}" do
+      edition = FactoryGirl.create(from, state: 'published')
+      sample_parts.each {|part| edition.parts.create(part)} if edition.respond_to?(:parts)
 
-    assert page.has_content?(answer.title)
-    assert page.has_content?("New edition created")
-    assert page.has_content?("Introductory paragraph")
-    assert page.has_content?("Will continue on")
-    assert page.has_content?("Link to start of transaction")
-    assert page.has_content?("More information")
+      visit_edition edition
 
-    within :css, '#edition_more_information_input' do
-      assert page.has_xpath?("//textarea[contains(text(), '#{answer.whole_body}')]"), "Expected to see: #{answer.whole_body}"
-    end
-  end
+      within "div.tabbable" do
+        click_on "Admin"
+      end
 
-  test "should be able to convert an AnswerEdition into a SimpleSmartAnswerEdition" do
-    answer = FactoryGirl.create(:answer_edition, state: 'published', body: "abcde")
-    visit_edition answer
+      select(to.to_s.gsub("_", " ").titleize.gsub(/Edition.*/, 'Edition'), from: 'to')
```

Comment:
`to.to_s.gsub("_", " ").titleize.gsub(/Edition.*/, 'Edition')` is hard to parse – I'd suggest having a helper method with a clear method name which calculates this.